app/models/move_event.rb

```diff
@@ -0,0 +1,7 @@
+class MoveEvent < Event
+  # NB: this class exposes a few methods specific to moves to the Event model
+
+  def to_location
+    @to_location ||= Location.find(event_params&.dig(:relationships, :to_location, :data, :id))
```

Comment:
It seems a little odd for a model to be delving into a params hash. I may be misunderstanding the use case here but possibly this is a controller or service responsibility to find the relevant location and persist against this model as `to_location_id` - it feels more idiomatic to then replace this with:

```ruby
  belongs_to :to_location, class_name: 'Location'
```

Reply:
lets chat